docs/source/tutorial/test_a_project.md

```diff
@@ -0,0 +1,335 @@
+│           │   nodes.py
+│           │   pipeline.py
+│
+tests
+|   ...
+└───pipelines
+    |    test_data_science_pipeline.py
+```
+
+Within `test_data_science.py` you can use [test classes](https://docs.pytest.org/en/7.1.x/getting-started.html#group-multiple-tests-in-a-class) to further organise your tests.
+
+### Using fixtures
+
+In our tests, we can see that `dummy_data` and `dummy_parameters` have been defined twice with the same values. Instead, we can define these outside of our tests as [pytest fixtures](https://docs.pytest.org/en/6.2.x/fixture.html#fixture):
+
+<details>
+
+```python
+import pytest
+
+@pytest.fixture
+def dummy_data():
+    return pd.DataFrame(
+        {"engines": [1, 2, 3],
+         "crew": [4, 5, 6],
+         "passenger_capacity": [5, 6, 7],
+         "price": [120, 290, 30]})
+
+@pytest.fixture
+def dummy_parameters():
+    parameters = {"model_options":
+                     {"test_size": 0.2,
+                      "random_state": 3,
+                      "features": ["engines", "passenger_capacity", "crew"]}
+                 }
+    return parameters
+```
+
+</details>
+
+We can then access these through the test arguments.
+
+```python
+def test_split_data(self, dummy_data, dummy_parameters):
+        ...
+```
+
+### Pipeline slicing
+
+In the test `test_data_science_pipeline` we test the that outputs of the data science pipeline are those that are expected. However, as pipelines are not static, this test is not robust. Instead we should be specific with how we define the pipeline to be tested; we do this by using [pipeline slicing](../nodes_and_pipelines/slice_a_pipeline.md#slice-a-pipeline-by-running-specified-nodes) to specify the pipeline's start and end:
+
+```python
+    def test_data_science_pipeline(self):
+        # Arrange pipeline
+        pipeline = create_pipeline().from_nodes("split_data_node").to_nodes("evaluate_model_node")
+        ...
+```
+
+This ensures that the test will still perform as designed, even with the addition of more nodes to the pipeline.
+
+
+After incorporating these testing practices, our test file `test_data_science.py` becomes:
+
+<details>
+
+```python
+# tests/pipelines/test_data_science_pipeline.py
+
+import pandas as pd
+import pytest
+
+from kedro.io import DataCatalog, MemoryDataset
+from kedro.runner import SequentialRunner
+from spaceflights.pipelines.data_science import create_pipeline
+from spaceflights.pipelines.data_science.nodes import split_data
+
+@pytest.fixture
+def dummy_data():
+    return pd.DataFrame(
+        {"engines": [1, 2, 3],
+         "crew": [4, 5, 6],
+         "passenger_capacity": [5, 6, 7],
+         "price": [120, 290, 30]})
+
+@pytest.fixture
+def dummy_parameters():
+    parameters = {"model_options":
+                     {"test_size": 0.2,
+                      "random_state": 3,
+                      "features": ["engines", "passenger_capacity", "crew"]}
+                 }
+    return parameters
+
+
+class TestDataScienceNodes:
```

Comment:
Can we _not_ use classes? 😄

Reply:
Is it poor practice? I was following the convention of the pre-existing example tests in our starters

Reply:
I don't think it's a bad practice per se but I think it's inherited from the old `unittest` (itself inspired in JUnit) way of doing things, and I think it's unnecessary